### PR TITLE
add set_each to button multistate variable

### PR DESCRIPTION
**Pull request by Astrovicis**

### CODE REVIEW

File: nanome/_internal/_ui/_button.py [hunk 1]
```diff
@@ -203,6 +203,13 @@ def _set_all(self, value):
             self._selected_highlighted = copy.deepcopy(value)
             self._unusable = copy.deepcopy(value)
 
+        def _set_each(self, **values):
+            self._idle = copy.deepcopy(values.get('idle')) or self._idle
+            self._highlighted = copy.deepcopy(values.get('highlighted')) or self._highlighted
+            self._selected = copy.deepcopy(values.get('selected')) or self._selected
+            self._selected_highlighted = copy.deepcopy(values.get('selected_highlighted')) or self._selected or self._selected_highlighted
```
Comment (ajm13) [inline nanome/_internal/_ui/_button.py:210]: `_selected` shouldn't take priority over `_selected_highlighted` here, same

File: nanome/api/ui/button.py [hunk 2]
```diff
@@ -405,6 +405,16 @@ def set_all(self, value):
             self.selected_highlighted = copy.deepcopy(value)
             self.unusable = copy.deepcopy(value)
 
+        def set_each(self, **values):
```
Comment (ajm13) [inline nanome/api/ui/button.py:408]: shouldn't this just call `self._set_all(values)`? same with `set_all` above tbh

File: nanome/_internal/_ui/_button.py (hunk 1 above)
Comment (Astrovicis) [reply nanome/_internal/_ui/_button.py]: ah, yes, you're right

File: nanome/api/ui/button.py (hunk 2 above)
Comment (Astrovicis) [reply nanome/api/ui/button.py:408]: documenting function overloads is...real hard. they do different things so I wanted to keep them separate

File: nanome/_internal/_ui/_button.py (hunk 1 above)
Comment (Astrovicis) [reply nanome/_internal/_ui/_button.py]: also I'm not convinced this definition needs to exist. I think maybe only button.py needs one

**Review by ajm13 — APPROVED** [review]

**Review by EthanVanderHorn — APPROVED** [review]

**Review by EthanVanderHorn — APPROVED** [review]

**Pull request merged**